cardano-cli/src/Cardano/CLI/Option/Flag.hs

```diff
@@ -0,0 +1,73 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE TypeOperators #-}
+
+module Cardano.CLI.Option.Flag
+  ( Flag (Flag)
+  , FlagOptions (FlagOptions)
+  , Defaultness (..)
+  , isDefault
+  , mkFlag
+  , parserFromFlags
+  )
+where
+
+import Cardano.CLI.Option.Flag.Types
+  ( Defaultness (IsDefault)
+  , Flag (Flag)
+  , FlagOptions (FlagOptions)
+  )
+import Cardano.CLI.Option.Flag.Types qualified as Z
+import Cardano.CLI.Vary
+import Cardano.CLI.Vary qualified as Vary
+
+import Control.Applicative
+import Data.Function
+import Options.Applicative (Parser)
+import Options.Applicative qualified as Opt
+
+-- | Create a parser from a help rendering function and list of flags.
+-- A default parser is included at the end of parser alternatives for
+-- the default flag (there should only be one default, but if more than
+-- one is specified, the first such one is used as the default).
+parserFromFlags :: (Flag a -> String) -> [Flag a] -> Parser a
+parserFromFlags _ [] = empty
+parserFromFlags mkHelp fs =
+  alternatives fs <|> defaults fs
+ where
+  alternatives [] = empty
+  alternatives (x : xs) =
+    parserFromFlag mkHelp x <|> alternatives xs
+
+  defaults :: [Flag a] -> Parser a
+  defaults [] = empty
+  defaults (x : xs) =
```

Comment:
I think you're missing default check here
```suggestion
  defaults (x : xs)
   | (Z.isDefault $ Z.options x) == Z.IsDefault = parserFromFlagDefault x
   | otherwise == defaults xs
```

By the way, `(Z.isDefault $ Z.options x) == Z.IsDefault` reads kind of weird. I think `Bool` usage would be justified here

Reply:
Thanks for the catch!